ui/components-react/src/components-react/newEditors/values/Values.ts

```diff
@@ -0,0 +1,126 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+/**
+ * Type definition for numeric value that can be handled by editor.
+ * @beta
+ */
+export interface NumericValue {
+  rawValue: number | undefined;
+  displayValue: string;
+  roundingError?: number;
+}
+
+/**
+ * Type definition for instance key value that can be handled by editor.
+ * @beta
+ */
+export interface InstanceKeyValue {
+  key: { id: string; className: string };
+  label: string;
+}
+
+/**
+ * Type definition for text value that can be handled by editor.
+ * @beta
+ */
+export interface TextValue {
+  value: string;
+}
+
+/**
+ * Type definition for boolean value that can be handled by editor.
+ * @beta
+ */
+export interface BooleanValue {
+  value: boolean;
+}
+
+/**
+ * Type definition for date value that can be handled by editor.
+ * @beta
+ */
+export interface DateValue {
+  value: Date;
+}
+
+/**
+ * Type definition for enum value that can be handled by editor.
+ * @beta
+ */
+export interface EnumValue {
+  choice: number | string;
+  label: string;
```

Comment:
does enum value need to have a label? I think it could be retrieved from metadata based on `choice`. Also, maybe `choice` should be renamed to `value`?

Reply:
Do we have cases where we render a property value without having its metadata? I think `InstanceKeyValue` is different, because getting its display value requires querying an iModel - looking at the metadata is not enough.

Reply:
I don't think we have such cases. I guess I chose to add label to the value to simplify value rendering when it's edited based on `PropertyRecord` that has `choices` as `Promise` https://github.com/iTwin/itwinjs-core/blob/1bc47d38777b51c4a3146c10207629defb38ee08/ui/appui-abstract/src/appui-abstract/properties/Description.ts#L32

However, I guess it will have to be awaited when creating `ValueMetadata` so it should not cause problems. Other use case I am now thinking is entering free form text when enum is not strict. Maybe in that case it would be useful to have label on value? For example if enum value is based on number and allows to enter label for that new choice.

Reply:
For non-strict enums, label always matches raw value. Of course, label is always a string, and value could be a number, but they easily convert to/from each other.